### PR TITLE
[mobile] Fix memory edge cases

### PR DESCRIPTION
**Pull request by AmanRajSinghMourya**
## Description
Handle edge cases in memory tap when tapping -
1. On 1st memory of the 1st memory block - reset the animation
2. On last memory of the last memory block - do nothing 

## Test

https://github.com/user-attachments/assets/3358bf41-548e-403b-83a2-aabac7a94266

### CODE REVIEW

File: mobile/apps/photos/lib/ui/home/memories/full_screen_memory.dart [hunk 1]
```diff
@@ -260,6 +260,9 @@ class _FullScreenMemoryState extends State<FullScreenMemory> {
       _onPageChange(inheritedData, currentIndex + 1);
     } else if (widget.onNextMemory != null) {
       widget.onNextMemory!();
+    } else {
+      hasFinalFileLoaded = true;
```
Comment (ashilkn) [inline mobile/apps/photos/lib/ui/home/memories/full_screen_memory.dart]: Let's use a different variable here? If I understood correctly, this is for making `_toggleAnimation` work. Instead of setting `hasFinalFileLoaded = true` here without actually knowing if final file has loaded, create a different state variable and use use `if (hasFinalFileLoaded || newVariable)` in `_toggleAnimation`.

File: mobile/apps/photos/lib/ui/home/memories/full_screen_memory.dart [hunk 2]
```diff
@@ -261,7 +262,7 @@ class _FullScreenMemoryState extends State<FullScreenMemory> {
     } else if (widget.onNextMemory != null) {
       widget.onNextMemory!();
     } else {
-      hasFinalFileLoaded = true;
+      isAtFirstOrLastFile = true;
```
Comment (ashilkn) [inline mobile/apps/photos/lib/ui/home/memories/full_screen_memory.dart:266]: Doesn't this have to be set to false later?

File: mobile/apps/photos/lib/ui/home/memories/full_screen_memory.dart [hunk 3]
```diff
@@ -271,10 +276,15 @@ class _FullScreenMemoryState extends State<FullScreenMemory> {
       _onPageChange(inheritedData, currentIndex - 1);
     } else if (widget.onPreviousMemory != null) {
       widget.onPreviousMemory!();
+    } else {
+      isAtFirstOrLastFile = true;
```
Comment (ashilkn) [inline mobile/apps/photos/lib/ui/home/memories/full_screen_memory.dart:280]: This is set to false immediately on `_onPageChange`, which negates the intent of setting it to true in the first place.

**Comment by ashilkn** [comment]
> On last memory of the last memory block - do nothing

@AmanRajSinghMourya Isn't closing the memory a better behaviour in this case? Or was this already discussed and decided to do nothing instead?

**Comment by AmanRajSinghMourya** [comment]
> > On last memory of the last memory block - do nothing
> 
> @AmanRajSinghMourya Isn't closing the memory a better behaviour in this case? Or was this already discussed and decided to do nothing instead?

It was decided to do nothing

**Review by ashilkn — APPROVED** [review]

**Review by ua741 — APPROVED** [review]
